src/modules/flow/iio/nodes.c

```diff
@@ -25,6 +25,16 @@
 
 #include <sol-iio.h>
 
+#define GEN_CHANNEL_NAME(_channel_name, _name, _error) \
+    if (channel_id >= 0) { \
```

Comment:
Oh, my bad. I forgot to mention that `channel_id` is also a good candidate to be a parameter. Idea is not depend on local variables from functions where this macro is called.